homeassistant/components/philips_air_purifier/fan.py

```diff
@@ -0,0 +1,166 @@
+        self._client.stop_observing_status(id(self))
+
+    def _set_status(self, status: Status | None) -> None:
+        self._status = status
+        self.schedule_update_ha_state()
+
+    @property
+    def unique_id(self):
+        """Return the unique ID for this purifier."""
+        if self._status is None:
+            return self._initial_data["unique_id"]
+        return self._status.device_id
+
+    @property
+    def name(self):
+        """Return the purifier's name."""
+        if self._status is None:
+            return self._initial_data["name"]
+        return self._status.name
+
+    @property
+    def available(self):
+        """Return whether the purifier is available."""
+        return self._status is not None
+
+    @property
+    def icon(self) -> str | None:
+        """Use an air purifier icon instead of the default fan."""
+        return "mdi:air-purifier"
+
+    @property
+    def device_info(self):
+        """
+        Return device information for the purifier.
+
+        Given that the purifier is sometimes not responsive for 30 seconds ore more,
+        we can't query it during startup and can only use data from the config entry.
+        """
+        info = {
+            "identifiers": {(DOMAIN, self.unique_id)},
+            "name": self.name,
+            "manufacturer": "Philips",
+            "model": self._initial_data["model"],
+        }
+        _LOGGER.debug("Device info: %s", str(info))
+        return info
+
+    @property
+    def is_on(self):
+        """Return whether the purifier is turned on, i.e. the fan is turning."""
+        return self._status.is_on
+
+    async def async_turn_on(
+        self,
+        percentage: int | None = None,
+        preset_mode: str | None = None,
+        **kwargs: Any,
+    ) -> None:
+        """Turn the purifier on with a given percentage or preset mode."""
+        await self._client.turn_on()
+
+    async def async_turn_off(
+        self,
+        **kwargs: Any,
+    ) -> None:
+        """Turn the purifier off."""
+        await self._client.turn_off()
+
+    @property
+    def preset_mode(self) -> str | None:
+        """Return the purifier's preset mode."""
+        if self._status is None:
+            return None
+        if self._status.mode == Mode.Manual:
+            return None
+        return self._status.mode.name
+
+    async def async_set_preset_mode(self, preset_mode: str) -> None:
+        """Activate a preset mode on the purifier."""
+        mode = Mode[preset_mode]
+        await self._client.set_preset_mode(mode)
+
+    device_to_speed_percentage_map = {
+        FanSpeed.Off: 0,
+        FanSpeed.Silent: 10,
+        FanSpeed.Speed1: 25,
+        FanSpeed.Speed2: 50,
+        FanSpeed.Speed3: 75,
+        FanSpeed.Turbo: 100,
+    }
+
+    @property
+    def percentage(self) -> int | None:
+        """Return the current fan speed converted to a percentage."""
+        if self._status is None:
+            return None
```

Comment:
I am not a huge fan of this self._status check everywhere. Can you see if this is required when you use `available` (see comment above)?

Reply:
You're right, Home Assistant doesn't seem to call the other methods if the device is unavailable. I've removed the checks everywhere except for the name, where the fallback to the value from the config entry makes sense, I think.

To make typing work, I added assertions, though (`self._status is not None`). Is this the way to go?